lib/permutations.rb

```diff
@@ -1,4 +1,31 @@
 
 def permutations?(string1, string2)
+  # take two strings as arguments and returns true if one string is a permutation of the other
   
+  hash = {}
+  string1.split("").each do |char|
+    if hash[char]
+      hash[char] += 1
+    else
+      hash[char] = 1
+    end
+  end
+  
+  puts hash
+  string2.split("").each do |char|
+    if ! hash[char]
+      return false
+    elsif hash[char] == 1
+      hash[char] = nil
+    elsif hash[char] > 1
+      hash[char] -= 1
+    end
+  end
```

Comment:
Generally speaking, it's good to avoid mixing the types used as keys or values in a given hash. I.e. a hash should typically have keys of a single type and values of a single type. In this case, mixing `nil` type with the `Integer` type can be avoided by letting the values decrease to zero, and then checking that all the values are zero on line 26.